server/graphql/programs/src/mutations/contact_trace/update.rs

```diff
@@ -0,0 +1,125 @@
+use async_graphql::*;
+use graphql_core::{
+    standard_graphql_error::{validate_auth, StandardGraphqlError},
+    ContextExt,
+};
+use repository::{contact_trace::ContactTraceFilter, StringFilter};
+use service::{
+    auth::{Resource, ResourceAccessRequest},
+    programs::contact_trace::upsert::{UpsertContactTrace, UpsertContactTraceError},
+};
+
+use crate::queries::contact_trace::ContactTraceNode;
+
+#[derive(InputObject)]
+pub struct UpdateContactTraceInput {
+    /// The patient ID the contact belongs to
+    pub patient_id: String,
+    /// The contact trace document type
+    pub r#type: String,
+    /// Contact trace document data
+    pub data: serde_json::Value,
+    /// The schema id used for the contact trace data
+    pub schema_id: String,
```

Comment:
Is this used anywhere ?

Reply:
Yes its the schema_id that has been used to edit the contact_trace. Think this is useful to record especially when schemas get updated over time.

Theoretically, a user can create a contact tracing entry with a certain schema while the system gets reconfigured to an incompatible schema. Providing the used schema_id makes sure that data isn't lost and the entry can still be updated.

Reply:
Ahh ok, i just traced it down, it's passed on to document, and then json_validator checks it.